src/ensemble_scheduler.cc

```diff
@@ -1032,18 +1041,28 @@ EnsembleContext::FinishEnsemble(std::unique_ptr<InferenceResponse>&& response)
   }
 
   if (ensemble_status_.IsOk()) {
-    if (info_->is_decoupled_) {
-      if (response != nullptr) {
-        InferenceResponse::Send(std::move(response), 0 /* flags */);
-      }
-      if (inflight_step_counter_ != 0) {
-        return ensemble_status_;
+    auto flags = (inflight_step_counter_ == 0)
```

Comment:
I am wondering of this `flags` is actually needed. We only use it on L1048. Should we also use it on L1064?